crates/bedrock/src/bedrock.rs

```diff
@@ -0,0 +1,133 @@
+mod models;
+
+use std::{str::FromStr};
+use std::any::Any;
+use anyhow::{Context, Result};
+use aws_sdk_bedrockruntime::types::{ContentBlockDeltaEvent, ContentBlockStartEvent, ConverseStreamMetadataEvent, ConverseStreamOutput, Message, MessageStartEvent, MessageStopEvent};
+use futures::{io::BufReader, stream::BoxStream, AsyncBufReadExt, AsyncReadExt, FutureExt, Stream, StreamExt};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use aws_sdk_bedrockruntime as bedrock;
+pub use aws_sdk_bedrockruntime as bedrock_client;
```

Comment:
I'll be honest, didn't think about that -- but, I don't believe so.
All the uses of tokio in the Bedrock SDK are opaque, so I don't know if they can be gotten around to use the gpui executors.

Probably to get around i'd have to implement SigV4 + use the raw HTTP client although it defeats the point of the SDK.

Reply:
In a similar vein I've had to export just the serde types and then construct the API calls inside of Zed. At the very least I wish that the [`HttpClient` trait](https://github.com/zed-industries/zed/blob/main/crates/http_client/src/http_client.rs#L39) was exported so that it would be easier to build libraries that [target GPUI](https://github.com/zed-industries/zed/blob/3901d4610115989d1b7e4d5c637a297da8219809/crates/gpui/src/app.rs#L733).

Reply:
Rust will only be happy with that if they're the same `HttpClient` between GPUI/Zed and the SDK config builder.